src/utils.js

```diff
@@ -5,3 +5,45 @@ export const useIsomorphicLayoutEffect = typeof window !== 'undefined' ? useLayo
 // useForceUpdate hook
 const forcedReducer = state => state + 1;
 export const useForceUpdate = () => useReducer(forcedReducer, 0)[1];
+
+export const createDeepProxy = (obj, affected) => {
+  const proxyMap = {};
+  const handler = {
+    get: (target, key) => {
+      if (!affected.has(target)) {
+        affected.set(target, [key]);
+      } else {
+        const used = affected.get(target);
+        if (!used.includes(key)) used.push(key);
+      }
+      const val = target[key];
+      if (typeof val !== 'object') {
+        return val;
+      }
+      if (!proxyMap[key]) {
+        proxyMap[key] = createDeepProxy(val, affected);
+      }
+      return proxyMap[key];
+    },
+  };
+  return new Proxy(obj, handler);
+};
+
+export const isDeepChanged = (origObj, nextObj, affected, cache, depth = 0) => {
+  if (origObj === nextObj) return false;
+  if (typeof origObj !== 'object') return true;
+  if (typeof nextObj !== 'object') return true;
+  if (!affected.has(origObj)) {
+    return depth !== 0; // false for root object, but true for others
```

Comment:
I'm wondering if this returns false always, we don't need to care about "rest" operator.

I really need help from @theKashey with this stuff...

Reply:
That might be needed in other cases. What if you `Object.keys(props)`?

Reply:
Oh, it's `Symbol(Symbol.iterator)`. I'm not sure about other cases, but what I could do is to assume the whole object is affected if accessed by a symbol property. (I hesitate writing my own proxy, when I understand that I don't really understand fully...)

Reply:
And, it's needed for reference equality checking too.

Could we ignore for the root object? Do people do this?
```javascript
  const state = useReduxState();
  return useMemo(() => ({ foo: 'foo' }), [state]);
```
Yeah, maybe?